src/main/java/org/hydev/mcpm/client/database/SuperLocalPluginTracker.java

```diff
@@ -0,0 +1,576 @@
+        String[] split = stringRepresentation.split(",");
+        this.name = split[0];
+        this.isManual = Boolean.parseBoolean(split[1]);
+        this.versionId = split[2];
+        this.pluginId = split[3];
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Boolean getManual() {
+        return isManual;
+    }
+
+    public String getVersionId() {
+        return versionId;
+    }
+
+    public String getPluginId() {
+        return pluginId;
+    }
+}
+
+/**
+ * This class keeps track of locally installed packages
+ *
+ * @author Kevin (https://github.com/kchprog)
+ * @since 2022-09-27
+ */
+public class SuperLocalPluginTracker implements SuperPluginTracker {
+    // CSV file storing the list of manually installed plugins.
+    // Now, each row in the csv file represents something as follows:
+    // ""name", "boolean", "versionId", "pluginId"
+    private String mainLockFile = "plugins/mcpm.lock.csv";
+
+    // Directory containing the plugins
+    private String pluginDirectory = "plugins";
+
+    /*
+     * Instantiates a LocalPluginTracker with default parameters for general use
+     */
+    public SuperLocalPluginTracker() {
+    }
+
+    /**
+     * Instantiates a LocalPluginTracker with custom parameters for testing or
+     *
+     * @param mainLockFile    The path to the main lock file
+     * @param pluginDirectory The path to the plugin directory
+     */
+    public SuperLocalPluginTracker(String mainLockFile, String pluginDirectory) {
+        this.mainLockFile = mainLockFile;
+        this.pluginDirectory = pluginDirectory;
+    }
+
+    /**
+     * Read metadata from a plugin's jar
+     *
+     * @param jar Local plugin jar path
+     * @return Metadata
+     */
+    private PluginYml readMeta(File jar) {
+        try (PluginJarFile InstancePluginJarFile = new PluginJarFile(jar)) {
+            return InstancePluginJarFile.readPluginYaml();
+        } catch (Exception e) {
+            System.out.println("Error reading plugin.yml from " + jar);
+            return null;
+        }
+    }
+
+    /**
+     * Read the CSV file and return a mapping between plugin names and install
+     * status
+     *
+     * @return Mapping between plugin name and boolean status
+     */
+    private ArrayList<PluginTrackerModel> readCsv() {
+        ArrayList<PluginTrackerModel> accumulator = new ArrayList<>();
+        try {
+            CSVReader reader = new CSVReader(new FileReader(mainLockFile));
+            String[] line;
+            while ((line = reader.readNext()) != null) {
+                accumulator.add(new PluginTrackerModel(line[0], Boolean.parseBoolean(line[1]), line[2], line[3]));
+            }
+            reader.close();
+            return accumulator;
+        } catch (FileNotFoundException e) {
+            return accumulator;
+        } catch (CsvException | IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Can you try using JSON instead of CSV? We're trying to remove the opencsv dependency because it's kind of large and one of its dependencies has known vulnerabilities

```java
// import static org.hydev.mcpm.Constants.JACKSON;
JACKSON.readValue(new File(mainLockFile), new TypeReference<>() {})
```